File: Moyen/ANEO_Sponsored_Puzzle/ANEO_Sponsored_Puzzle.java
import java.util.*;

class Solution {

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        int speed = in.nextInt();
        int lightCount = in.nextInt();
        ArrayList<Integer> tDistance = new ArrayList<>();
        ArrayList<Integer> tDuration = new ArrayList<>();
        for (int i = 0; i < lightCount; i++) {
            tDistance.add(in.nextInt());
            tDuration.add(in.nextInt());
        }
        int compteur = 0;
        int newSpeed = speed;
        while (lightCount != compteur) {
            compteur = 0;
            for (int k=0;k<tDuration.size();k++) {
                if (Math.floor(newSpeed*tDuration.get(k)/3.6) > tDistance.get(k) || 
                    Math.floor(newSpeed*tDuration.get(k)/3.6)%tDistance.get(k) == 0 || 
                    Math.floor(tDistance.get(k)/(newSpeed*tDuration.get(k)/3.6))%2 == 0 || 
                    Math.floor(newSpeed*tDuration.get(k)/3.6) == Math.floor(tDistance.get(k)%(newSpeed*tDuration.get(k)/3.6))) {
                    compteur++;
                }
                if (lightCount == compteur) {
                    System.out.println(newSpeed);
                }
            }
            newSpeed--;
        }
        in.close();
    }
}

File: Moyen/ANEO_Sponsored_Puzzle/ANEO_Sponsored_Puzzle.js
const speed = parseInt(readline());
const lightCount = parseInt(readline());
let tabDuration = [];
let tabDistance = [];
for (let i = 0; i < lightCount; i++) {
    var inputs = readline().split(' '); 
    tabDuration[i] = parseInt(inputs[1]);
    tabDistance[i] = parseInt(inputs[0]);
}

let newSpeed = speed;
let compteur;
while (lightCount != compteur) {
    compteur = 0;
    for (let k=0;k<tabDuration.length;k++) {
        if (Math.floor(newSpeed*tabDuration[k]/3.6) > tabDistance[k] || 
            Math.floor(newSpeed*tabDuration[k]/3.6)%tabDistance[k] === 0 || 
            Math.floor( tabDistance[k]/ (newSpeed*tabDuration[k]/3.6) )%2 === 0 || 
            Math.floor(newSpeed*tabDuration[k]/3.6) === Math.floor(tabDistance[k]%(newSpeed*tabDuration[k]/3.6))) {
                compteur++;
        }
        if (lightCount === compteur) {
            console.log(newSpeed);
        }
    }
    newSpeed--;
}